Improve navigation mesh code examples (#8714)
godotengine/godot-docs: tutorials/navigation/navigation_using_navigationmeshes.rst

    Remove the geometry that is on the ground inside the other geometry. If that is not possible, add smaller "dummy" geometry inside with as few triangles as possible so the cells
    are occupied with something.

Navigation mesh script templates
--------------------------------

> The following script uses the NavigationServer to parse source geometry from the SceneTree, bakes a navigation mesh, and updates a navigation region with the updated navigation mesh.

Reviewer: ```suggestion
The following script uses the NavigationServer to parse source geometry from the scene tree, bakes a navigation mesh, and updates a navigation region with the updated navigation mesh.
```

It's not related to the `SceneTree` class, but about the scene tree concept.

.. tabs::
 .. code-tab:: gdscript 2D GDScript

    extends Node2D

    var navigation_mesh: NavigationPolygon
    var source_geometry : NavigationMeshSourceGeometryData2D
    var callback_parsing : Callable
    var callback_baking : Callable
    var region_rid: RID

    func _ready() -> void:
        navigation_mesh = NavigationPolygon.new()
        navigation_mesh.agent_radius = 10.0
        source_geometry = NavigationMeshSourceGeometryData2D.new()

>         callback_parsing = Callable(self, "on_parsing_done")
>         callback_baking = Callable(self, "on_baking_done")

Reviewer: ```suggestion
        callback_parsing = on_parsing_done
        callback_baking = on_baking_done
```

Method is itself a Callable.

        region_rid = NavigationServer2D.region_create()

        # Enable the region and set it to the default navigation map.
        NavigationServer2D.region_set_enabled(region_rid, true)
        NavigationServer2D.region_set_map(region_rid, get_world_2d().get_navigation_map())

        # Some mega-nodes like TileMap are often not ready on the first frame.
        # Also the parsing needs to happen on the main-thread.
        # So do a deferred call to avoid common parsing issues.

>         call_deferred("parse_source_geometry")

Reviewer: ```suggestion
        parse_source_geometry.call_deferred()
```

Same here. Better avoid using strings :)

        # so the obstructions outlines have something to "cut" into.
        source_geometry.add_traversable_outline(PackedVector2Array([
            Vector2(0.0, 0.0),
            Vector2(500.0, 0.0),
            Vector2(500.0, 500.0),
            Vector2(0.0, 500.0)
        ]))

        # Bake the navigation mesh on a thread with the source geometry data.
        NavigationServer2D.bake_from_source_geometry_data_async(
            navigation_mesh,
            source_geometry,
            callback_baking
        )

    func on_baking_done() -> void:
        # Update the region with the updated navigation mesh.
        NavigationServer2D.region_set_navigation_polygon(region_rid, navigation_mesh)

 .. code-tab:: gdscript 3D GDScript

    extends Node3D

    var navigation_mesh: NavigationMesh
    var source_geometry : NavigationMeshSourceGeometryData3D
    var callback_parsing : Callable
    var callback_baking : Callable
    var region_rid: RID

    func _ready() -> void:
        navigation_mesh = NavigationMesh.new()
        navigation_mesh.agent_radius = 0.5
        source_geometry = NavigationMeshSourceGeometryData3D.new()

>         callback_parsing = Callable(self, "on_parsing_done")
>         callback_baking = Callable(self, "on_baking_done")

Reviewer: ```suggestion
        callback_parsing = on_parsing_done
        callback_baking = on_baking_done
```

Same as the 2D version.

        region_rid = NavigationServer3D.region_create()

        # Enable the region and set it to the default navigation map.
        NavigationServer3D.region_set_enabled(region_rid, true)
        NavigationServer3D.region_set_map(region_rid, get_world_3d().get_navigation_map())

        # Some mega-nodes like GridMap are often not ready on the first frame.
        # Also the parsing needs to happen on the main-thread.
        # So do a deferred call to avoid common parsing issues.

>         parse_source_geometry.call_deferred()

Reviewer 2: This, specifically. Again, better to document this than not, but maybe something we should file somewhere to see if we can improve UX here.

>         call_deferred("parse_source_geometry")

Reviewer: ```suggestion
        parse_source_geometry.call_deferred()
```

Same as the 2D version.
